infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInfraFactory.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.workspace.infrastructure.docker;
+
+import com.google.inject.assistedinject.Assisted;
+
+import org.eclipse.che.api.core.ValidationException;
+import org.eclipse.che.api.core.model.workspace.config.Environment;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.docker.model.DockerEnvironment;
+
+import java.util.List;
+
+/**
+ * @author Alexander Garagatyi
+ */
+public interface DockerInfraFactory {
```

Comment:
I would say that this name is not a good one and may be confusing, while `DockerRuntimeFactory` is better candidate.
Can you please add a small description which describes a purpose of this interface(solve dependency routine i think).
Does it make sense to make this interface package private, so user will not use it for `Runtime` instances creation?
Does it make sense to have separate factories for context & runtime?